### PR TITLE
clone the service under lock to avoid a data race

### PR DESCRIPTION
**Pull request by dhiaayachi**
This is a fix to the data race discovered in [here](https://app.circleci.com/pipelines/github/hashicorp/consul/25488/workflows/43e31311-9cf6-43d3-9eef-0eaef96be7ad/jobs/540392/tests)

### CODE REVIEW

**Review by freddygv — APPROVED** [review]
LGTM

File: agent/local/state.go [hunk 1]
```diff
@@ -1082,27 +1082,29 @@ func (l *State) updateSyncState() error {
 			continue
 		}
 
+		svc := *ls.Service
```
Comment (markan) [inline agent/local/state.go:1085]: Might be worth explaining why this is done so someone doesn't come by and 'optimize' it out.

**Review by markan — APPROVED** [review]
Other than potential comment LGTM

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/542131.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 1eac39ae9c290431faf71e396559dc896d453436 onto `release/1.11.x` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 1eac39ae9c290431faf71e396559dc896d453436 onto `release/1.10.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/542131)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 1eac39ae9c290431faf71e396559dc896d453436 onto `release/1.9.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/542131)

File: agent/local/state.go [hunk 2]
```diff
@@ -1082,27 +1082,34 @@ func (l *State) updateSyncState() error {
...
+		if !reflect.DeepEqual(svc.TaggedAddresses, rs.TaggedAddresses) {
+			if svc.TaggedAddresses == nil {
+				svc.TaggedAddresses = make(map[string]structs.ServiceAddress)
 			}
 			for k, v := range rs.TaggedAddresses {
 				if strings.HasPrefix(k, structs.MetaKeyReservedPrefix) {
-					ls.Service.TaggedAddresses[k] = v
+					svc.TaggedAddresses[k] = v
 				}
 			}
```
Comment (kisunji) [inline agent/local/state.go:1107]: `svc.TaggedAddresses` should be cloned before inserting new values or else the map of the original *Service also gets mutated ([example](https://go.dev/play/p/NPc8FBjEeWI))

**Pull request merged**